projects/swimlane/ngx-ui/src/lib/components/button/file-button.component.ts

```diff
@@ -111,20 +69,24 @@ export class FileButtonComponent implements OnInit {
       'standard-style': this.styleType === FileButtonStyleType.standard,
       'progress-style': this.styleType === FileButtonStyleType.progress,
       'show-progress': this.uploader && this.uploader.options.isHTML5,
-      success: this.isItemSuccessful,
+      success: this._isItemSuccessful,
       active: this.uploader && this.uploader.isUploading
     };
   }
 
-  isItemSuccessful: boolean = false;
-  progress: string = '0%';
+  readonly FileButtonStyleType = FileButtonStyleType;
+  progress: number = 0;
```

Comment:
not too big a deal since we're under construction but this would be a breaking change right? if someone consuming it was expecting the emitter to emit a string

Reply:
Ya it could, but we are the lib and there are going to be times when the inputs/outputs might need to change for components, we just have to be aware of it when we integrate it into platform and turbine frontends.

Reply:
yup, and i like this change as well. ship it 🚀